test/unit/org/apache/cassandra/metrics/CacheMetricsTest.java

```diff
@@ -38,10 +39,23 @@
     private static final long capacity = 65536;
 
     @Test
+    public void testCodahaleCacheMetrics()
+    {
+        System.setProperty("cassandra.use_micrometer_metrics", "false");
```

Comment:
should we clear `cassandra.use_micrometer_metrics` in `@AfterClass` ?

Reply:
I totally sympathize with the cause of being able to run unit tests within the same JVM, but I also believe it is a lost cause :(
E.g. in metrics many tests use embedded Cassandra daemon, but IIUC there is no way to clean up after it, so two such tests cannot be run in a single JVM. One effectively must fork jvm for each test, which makes cleaning system properties moot.

That said, I fixed the property cleanup in this and other tests.